### DISCUSSION TITLE
how to update helix-editor on Ubuntu?

### DISCUSSION THREAD

**Discussion by matan-h** [helix-editor/helix#2069] [Q&A] [answered]
I have Ubuntu so to install I do  `git clone` , and build it from source. (because helix isn't released to ppa,snap or [flatpak](#1968 )  )
and when the code becomes outdated -  do a `git pull` , reset the runtime folder to the head^1 and rebuild it from source (
```bash
git pull 
git reset --hard HEAD^ # reset the runtime folder (and all project with it)
cargo install --path helix-term 
hx --grammar fetch 
hx --grammar build 
[[ -L ~/.config/helix/runtime && -d ~/.config/helix/runtime ]] || cp -r runtime ~/.config/helix/runtime # copy if the file is not symlink
```
).

Is there better way to update?

**Comment by the-mikedavis** [comment, marked as answer]
A symlink from `~/.config/helix/runtime` to the runtime directory in the source makes upgrades easier. For example if you checked the repository out like so:

```sh
$ cd ~/src                                          # say you check stuff out into ~/src
$ git clone git@github.com:helix-editor/helix.git
$ mkdir -p ~/.config/helix
$ pushd ~/.config/helix
$ ln -s ~/src/helix/runtime .                       # setup the symlink
$ popd
$ cargo install --path helix-term                   # initial installation
```

Then when you'd like to update,

```sh
$ cd ~/src/helix
$ git pull
$ cargo install --path helix-term
```

Then you don't need to do anything with the runtime directory because of the symlink.

**Reply by matan-h** [reply]
I still need to do this commands from `Installation` in the readme?
 ```bash
git reset --hard HEAD^ # reset the runtime folder (and all project with it) for refetch and rebuild
hx --grammar fetch
hx --grammar build
```

**Reply by the-mikedavis** [reply]
If you use a symlink between the source and the runtime directory, no. `cargo install` will fetch and build the grammars itself.
